oci/dir.go

```diff
@@ -0,0 +1,142 @@
+package oci
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/opencontainers/go-digest"
+)
+
+// Directory is ImageDriver for directory representation of an OCI image.
+// The directory must not exist before calling Init function of this structure.
+func Directory(path string) ImageDriver {
```

Comment:
Just note that this directory `ImageDriver` is not used at all in containerd, but can be used for 3rd party projects.
(My https://github.com/AkihiroSuda/filegrain is going to vendor this)